contribs/gnodev/pkg/dev/node.go

```diff
@@ -412,6 +427,39 @@ func (n *Node) rebuildNodeFromState(ctx context.Context) error {
 	return nil
 }
 
+func (n *Node) handleEventTX(evt tm2events.Event) {
+	switch data := evt.(type) {
+	case bft.EventTx:
+		go func() {
+			// Use separate routine to avoid deadlock
```

Comment:
I find this confusing as someone reading the code. Does this really mean deadlock or does it mean blocking? I see that there are many places in this file with this pattern:
```go
n.muNode.Lock()
defer n.muNode.Unlock()
```
So why is this particular case different?  A clarifying comment could go a long way here.

Reply:
Sure, this is because the callback is called while rebuilding the node. During this phase, the node holds the lock, so without this goroutine, the callback will indefinitely wait for the lock to be released.

Reply:
I will improve the comment